Automattic/simplenote-ios
# Keychain Migration

Here's an attempt at migrating the token that is stored in the keychain to the new keychain.

This is pretty hard to test, but I was able to see it working by:

* Installing Simplenote from the app store and signing in on my test device.
* Delete the share target :)
* Changing the bundle identifier to `com.codality.NotationalFlow.MigrationDev`
* Set the UserDefault to the username at the beginning of the `migrateIfNecessary()` method: `UserDefaults.standard.set("{your-username", forKey:usernameKey)`
* Run the app to your device and you should be signed in (after dismissing the welcome view because the app thinks it is the first run)

This all hinges on the `SPUsername` user default persisting across the app updates, so we need to test that!
